src/plugins/content_management/public/content_client/content_client.tsx

```diff
@@ -9,13 +9,16 @@
 import { QueryClient } from '@tanstack/react-query';
 import { createQueryObservable } from './query_observable';
 import type { CrudClient } from '../crud_client';
-import type { CreateIn, GetIn } from '../../common';
+import type { CreateIn, GetIn, UpdateIn, DeleteIn, SearchIn, SearchOut } from '../../common';
 
 const queryKeyBuilder = {
   all: (type: string) => [type] as const,
   item: (type: string, id: string) => {
     return [...queryKeyBuilder.all(type), id] as const;
   },
+  search: (type: string, params: unknown) => {
```

Comment:
> Can we call it query instead of params to align with the rpc query naming?

Sure don't mind, I so you already did that in your pr. 

> Also, can we now rename type with id as we use id to uniquely identify a ContentType ?

Hm, ContentTypeRegistry id means id of the type. But here the main subject is Content (Content item itself), so id refer to the content. e.g. see item which is a key for get: 

item: (type: string, id: string) => {
    return [...queryKeyBuilder.all(type), id] as const;
  },



maybe it should be? 

item: (contentTypeId: string, itemId: string) => {
    return [...queryKeyBuilder.all(contentTypeId), itemId] as const;
  },


search: (contentTypeId: string, query: unknown) => {

Reply:
Yes, server side I am using `contentTypeId`, it seems more consistent to refer to `ContentType.id`.